platform/aws.go

```diff
@@ -32,6 +32,12 @@ type awsMachine struct {
 	mach    *ec2.Instance
 }
 
+var _ Machine = (*awsMachine)(nil)
```

Comment:
Considering Go tends to not use explicit assertions I think the implicit one in NewMachine is perfectly sufficient.